Add/drag handle

oandregal opened the pull request:
This PR adds the drag handle to the up/down mover.

![peek 2018-09-03 20-02](https://user-images.githubusercontent.com/583546/44998257-56cc3e00-afb4-11e8-9d4f-8ae7e464a093.gif)

### TODO

- [x] Change icon to drag state.
- [x] Add current `BlockDraggable` conditions.
- [x] Call block's `onDragStart` / `onDragEnd`.
- [x] Add `isDragHandleVisible` prop.
- [x] Add aria roles.
- [x] Fix overflow for one-liner paragraph blocks.
- [x] Update tests.
- [x] Show drag handle while dragging.
- [x] Hide classic toolbar while dragging.
- [x] Refactor `BlockDraggable`?

### Test

Regular drag-n-drop operation (including child blocks, and embeds).

oandregal committed:
Add drag handle icon to mover component

oandregal committed:
Add drag handle to block mover

mtias commented on packages/editor/src/components/block-mover/index.js:
```diff
@@ -66,6 +66,30 @@ export class BlockMover extends Component {
 					onFocus={ this.onFocus }
 					onBlur={ this.onBlur }
 				/>
+				<Draggable
```
Thinking it might be worth setting this as an optional prop (default `draggable`  to true) on the block mover.

mtias replied:
Yes, I was thinking something that would let you use the mover without drag handles for a certain UI if needed.

mtias commented on packages/editor/src/components/block-mover/index.js:
```diff
@@ -66,6 +66,30 @@ export class BlockMover extends Component {
+				>
+					{
+						( { onDraggableStart, onDraggableEnd } ) => (
+							<IconButton
```
Not sure we should use the button styles here. It looks a bit weird on hover.

mtias replied:
I think the space should also be a bit tighter overall.

nosolosw replied:
I'll leave @jasmussen apply his svg / css :magician: skills while I watch from the distance and learn.

oandregal committed:
Extract to own component

oandregal committed:
Hide dragHandle if block cannot be dragged

oandregal committed:
Call block onDragStar/onDragEnd methods

oandregal committed:
Do not use BlockDraggable

oandregal committed:
Change cursor to drag on the drag handle

oandregal committed:
Remove no longer necessary CSS

oandregal committed:
On dragging, set block opacity to 0.7

We no longer have a dedicated and invisible div for the drag handle
that we show on dragging and acts as a grey box on top of the block
being dragged.

nosolosw commented:
Also for design feedback: note that this PR also changes how Gutenberg indicates which block is being dragged: it uses the block's opacity instead of showing a grey box above the block.

In the spirit of doing one thing at a time, I wouldn't have changed that behavior in this PR if it wasn't for this: the grey box was a dedicated div that acted as a drag handle and that was made visible when the dragging process started; now that we've removed the dedicated drag handle we no longer have that div node to play with. Also: the opacity seems nicer to me in this case. If we still consider the grey box a preferred UI, I can figure out how to maintain that behavior.

Commit:
Polish drag handle and metrics

This makes the drag handle and up/down buttons slightly smaller, and centers them.

jasmussen commented:
Very nice work here. 

I pushed a little polish in https://github.com/WordPress/gutenberg/pull/9569/commits/4128f21beebae8bf1a6071704c371b8fdec3dceb:

<img width="370" alt="screen shot 2018-09-04 at 10 26 36" src="https://user-images.githubusercontent.com/1204802/45019599-0d6c0500-b02d-11e8-80e6-b198f9da9486.png">

That is, buttons are now slightly smaller so as to better fit on a single paragraph height, which is the smallest block we can have. The paragraph is actually 56px tall, and the buttons are now 72px (3*24px). So I've compensated by vertically centering them. Here's with a little debug help:

<img width="235" alt="screen shot 2018-09-04 at 10 19 06" src="https://user-images.githubusercontent.com/1204802/45019645-2bd20080-b02d-11e8-9cd7-629ae37cbd07.png">

This is before:

<img width="236" alt="screen shot 2018-09-04 at 09 55 46" src="https://user-images.githubusercontent.com/1204802/45019652-2f658780-b02d-11e8-85b7-76212379c975.png">

I'm going to next take a stab at tweaking the hover style of the drag handle. It shouldn't be a button, the hand cursor itself should be _mostly_ sufficient, perhaps with a color change on hover. 

> Also for design feedback: note that this PR also changes how Gutenberg indicates which block is being dragged: it uses the block's opacity instead of showing a grey box above the block

I think it's okay to visit this, it's a bit challenged as is.

The thing is, you can now style the editor... you can change the text color and the background color using editor styles. But unless @youknowriad with his new CSS parser can do magic here, we can't know which background color to show on a block that's being dragged. Until now we've simply assumed white.

Ideally we'd know what the color would be, it's a little hard to see that you're dragging a block right now since it has no background. But I think it's good to surface this discussion on what to do here. 

Another option is, instead of showing a clone of the block, showing a small _icon_ of the block. Like, imagine as you "lift the block off the page", it transforms into a small block next to your cursor, and expands again when placed. That way we'd cover less of the page. But I know @mtias has some opinions here we should listen to.

Commit:
Try: Remove hover style and make un-tabbable

This needs a sanity check and is a separate commit for that reason, so easier to revert if it isn't kosher.

This commit removes the hover and focus styles for the drag handle.

It also zeroes out the aria label, makes aria hidden, and unfocusable using the keyboard. The idea being that you cannot at all interact with the drag handle using just the keyboard, we have the movers for that, and thus this item shouldn't be announced to either screen readers or be part of the tab index as that's simply useless.

Thoughts?

jasmussen commented:
In https://github.com/WordPress/gutenberg/pull/9569/commits/8c575d207c258f6710d8f5aa93293d6c61b0fd2a I committed a change I'd like a sanity check on. From the commit message:

<blockquote>
Try: Remove hover style and make un-tabbable

This needs a sanity check and is a separate commit for that reason, so easier to revert if it isn't kosher.

This commit removes the hover and focus styles for the drag handle.

It also zeroes out the aria label, makes aria hidden, and unfocusable using the keyboard. The idea being that you cannot at all interact with the drag handle using just the keyboard, we have the movers for that, and thus this item shouldn't be announced to either screen readers or be part of the tab index as that's simply useless.

Thoughts?
</blockquote>

Here's how this looks when mousing:

![dragging now](https://user-images.githubusercontent.com/1204802/45021362-a69d1a80-b031-11e8-9b5f-fd408f5bacb6.gif)

Here's how this looks when keyboarding:

![tabbing](https://user-images.githubusercontent.com/1204802/45021374-adc42880-b031-11e8-8521-84e6cd7012ed.gif)

Note by the way that in Chrome on Mac I'm seeing [a "copy" cursor](https://support.apple.com/kb/PH25689?locale=en_US) as soon as I start dragging. I should be seeing the `grabbing` cursor instead. Any insights as to why that might be?

jasmussen commented:
> Note by the way that in Chrome on Mac I'm seeing a "copy" cursor as soon as I start dragging. I should be seeing the grabbing cursor instead. Any insights as to why that might be?

Some further info on this, the copy cursor is supposed to show up in the file system while you're dragging a file and holding ⌥ at the same time. 

Ironically, I'm experiencing the inverse behavior both in master and in this branch: when I'm holding ⌥, the copy cursor disappears and the correct `grabbing` cursor is present. 

The behavior is correct in Safari — when I drag normally, I'm seeing the correct `grabbing` cursor. When I hold ⌥ in Safari, I'm seeing the copy cursor. 

So it seems as though Chrome is either buggy, or my system is buggy, or some part of our code makes Chrome think that by default a drag action is a _copy/duplicate_ action as opposed to a _move_ action.

oandregal committed:
Set the drag/drop effect to move

oandregal committed:
Revert "Set the drag/drop effect to move"

This reverts commit a96bb4b621c608a2534bbb3b8b2198830500dfa8.

It was a test to see if chrome on mac would show the hand
cursor while dragging. It doesn't. It shows the default icon.

nosolosw commented:
In a96bb4b we experimented with setting the dnd event to `move` to see if that affected how chrome on mac styled things. It seems it does. It shows the default cursor instead:

![cursor](https://user-images.githubusercontent.com/583546/45023064-f67de080-b035-11e8-9d50-27d27f2e4a25.gif)

Change reverted in 3e595c5.

Commit:
Remove hover color, and make bg color friendly.

jasmussen commented:
Removed the hover color from the drag handle, and also made it bg color friendly. 
<img width="132" alt="screen shot 2018-09-04 at 11 35 17" src="https://user-images.githubusercontent.com/1204802/45023389-a6534e00-b036-11e8-97ad-2a96f1c891ca.png">

nosolosw commented:
Created an issue to track the cursor issue at https://github.com/WordPress/gutenberg/issues/9610

afercia commented:
The `focusable` attribute rendered in the HTML is invalid HTML and should be omitted:
> Error: Attribute focusable not allowed on element button at this point.

Commit:
Remove focusable.

jasmussen commented:
> The focusable attribute rendered in the HTML is invalid HTML and should be omitted:

Good catch, thanks. Is this implicit approval of the assertion that the drag handle should not be part of the tab order/be keyboardable?

afercia commented:
Yup, I'd agree taking it out from the tab sequence and hiding it with `aria-hidden=true` makes sense, as it can't be used with a keyboard.

oandregal committed:
Use a div as the drag handle

Firefox can't make buttons draggable.

oandregal committed:
Add tooltip

nosolosw commented:
@jasmussen in 8ee6afa I've changed the `IconButton` (which ended up adding a `button` element) by a simple `div` for the drag handle. The reason is: Firefox won't make buttons draggable. Would you mind giving it a sanity check and verify if it unintentionally broke any of your modifications?

oandregal committed:
Expose an isDraggable property to control draggability

So drag is something that can be turned on/off depending
on different UI needs.

afercia commented:
> in 8ee6afa I've changed the IconButton (which ended up adding a button element) by a simple div

Then, I'd suggest to remove `tabIndex`, as a div is not focusable by default. Also `onFocus / onBlur` I guess. And the empty `aria-label=""` is a bit pointless.

Commit:
Remove empty aria label and tabindex.

jasmussen commented:
I removed the aria label and tabindex.

I would honestly prefer we remove the tooltip as well — the drag handle feels obvious to me, and because it is an _addition for mouse users_ to the clickable up/down arrows, it doesn't feel necessary to me. 

On the flipside, it feels intrusive, and often invokes even when it shouldn't:

<img width="973" alt="screen shot 2018-09-06 at 08 44 01" src="https://user-images.githubusercontent.com/1204802/45139805-49c86e00-b1b1-11e8-8bea-72eb5534d7a5.png">

— at this point I'm dragging, so the tooltip should disappear. In other words, if we are to have a tooltip, we need to make sure it disappears as soon as you start dragging.

I also feel like we should consider doing something about the transparent background while dragging. I'm going to try something, but not sure it'll work.

afercia commented:
Not sure why `onFocus` and `onBlur` are kept, as this `<div>` can't be receive focus and can't be "blurred".

oandregal committed:
Update tests

oandregal committed:
Remove tooltip

Commit:
Try a different style for moving.

jasmussen commented:
@nosolosw I'll let you respond to the onfocus thoughts. 

For now, I've made a change to the visual styling of the dragondrop:

![changes](https://user-images.githubusercontent.com/1204802/45145325-aaab7280-b1c0-11e8-9943-60fa6a9f7597.gif)

I know this is kind of like how it used to be. I'd like to explain why it is this way. Most importantly: we are not _cloning_ a block when you are dragging, we are _moving_, so it's important the block you move is not seen as a clone. Hence the gray BG. 

Also, I hid the block toolbar and mover on the "block that's left behind", it didn't make sense to keep those there. But I am hiding them on the dragged block.

I tried unhiding the editor-block-mover on the draggable clone — because otherwise the grip you're holding disappears when you are dragondropping, that's a little weird. But doing so makes dragondrop not work. Any idea why?

Commit:
Make the gray spot only fit the content.

oandregal committed:
Remove unnecessary blur/focus handlers

nosolosw commented:
Removed the `onFocus` / `onBlur` and the Tooltip. Being the drag handler not focusable it doesn't make a lot of sense to have those.

nosolosw commented:
> I tried unhiding the editor-block-mover on the draggable clone — because otherwise the grip you're holding disappears when you are dragondropping, that's a little weird. But doing so makes dragondrop not work. Any idea why?

I'll take a look. Perhaps a race condition: hidden nodes won't take events, so depending on how we do that (before/after connecting the event) the visibility change may be messing up with the DnD event.

jasmussen commented:
> I'll take a look. Perhaps a race condition: hidden nodes won't take events, so depending on how we do that (before/after connecting the event) the visibility change may be messing up with the DnD event.

I tried adding `pointer-events: none;` and even `z-index: 0` to make the mover/handle visible on the drag clone even if not interactable. But this didn't work.

oandregal committed:
Remove no longer necessary code

oandregal committed:
Show the mover in the clone image

The drag handle is part of the mover so we want to keep it visible
while dragging.

jasmussen commented on packages/editor/src/components/block-list/style.scss:
```diff
@@ -2,25 +2,12 @@
-	// Nested blocks don't have any side affordance for drag and drop
-	.editor-block-list__layout &,
-	> * {
-		pointer-events: auto;
```
🎉

nosolosw replied:
_Adds a feature by removing code_

oandregal committed:
Use BlockDraggable instead of Draggable

This makes BlockDraggable DOM agnostic and uses it as a base to
compose drag data. The vision is that we may have an ImageDraggable
component in the future that would abstract what data the dragging
operation needs to drag images, and so on.

The goal would be to use it by only providing one prop, the
blockClientId.

Making this a single commit in case we want to revert.

nosolosw commented on edit-post/assets/stylesheets/_z-index.scss:
```diff
@@ -33,9 +33,6 @@ $z-layers: (
 	".components-button.is-button {:focus or .is-primary}": 1,
 
-	// Should have lower index than anything else positioned inside the block containers
-	".editor-block-list__block-draggable": 0,
```
Dragging no longer needs a dedicated DOM node. This PR removes the `editor-block-list__block-draggable` node from  `BlockDraggable`.

nosolosw commented on packages/editor/src/components/block-draggable/index.js:
```diff
@@ -0,0 +1,34 @@
+ */
+import { Draggable } from '@wordpress/components';
+
+const BlockDraggable = ( { children, clientId, rootClientId, blockElementId, layout, order, onDragStart, onDragEnd } ) => {
```
The vision for this component is that we should only provide one prop that identifies the block to be dragged. The rest (except for the optional `onDragStart` and `onDragEnd` event handlers) should be derived or retrieved from the state.

I wonder whether at this stage this is useful or just introduces more indirection with no real benefits. We may want to remove it completely in favor of `Draggable` at the moment.

youknowriad replied:
Agreed, it doesn't add a lot of value at the moment especially since it's used only once but I have a small preference for keeping it. We can make it more useful. For instance the `rootClientId` and `order` props can already be added using `withSelect` and avoid passing them from above.

nosolosw commented on packages/editor/src/components/block-mover/index.js:
```diff
@@ -18,7 +18,8 @@ import { withInstanceId, compose } from '@wordpress/compose';
  */
 import { getBlockMoverDescription } from './mover-description';
-import { upArrow, downArrow } from './arrows';
+import { upArrow, downArrow, dragHandle } from './arrows';
```
Perhaps the `dragHandle` icon should be declared within the `drag-handle` component?

jasmussen commented:
I fixed the classic block, and there's a followup PR that makes things prettier also in #9680.

youknowriad commented on packages/editor/src/components/block-list/style.scss:
```diff
@@ -1,104 +1,40 @@
-		.editor-block-list__layout & {
-			left: 0;
-			right: 0;
+		.freeform-toolbar,
```
Why targetting a specific block. Maybe these could be moved to the `editor.scss` file of the block?

youknowriad commented on packages/editor/src/components/block-mover/arrows.js:
```diff
@@ -14,3 +14,11 @@ export const downArrow = (
 		<polygon points="9,13.5 14.7,7.9 13.2,6.5 9,10.7 4.8,6.5 3.3,7.9 " />
 	</AccessibleSVG>
 );
+
```
Should we rename this file `icons` instead of `arrows`?

nosolosw replied:
e9dc8aae6a01f34b01c771105ff024600402cd08

youknowriad commented on packages/editor/src/components/block-mover/drag-handle.js:
```diff
@@ -0,0 +1,42 @@
+ */
+import BlockDraggable from '../block-draggable';
+
+export const IconDragHandle = ( { isVisible, className, icon, onDragStart, onDragEnd, blockElementId, order, rootClientId, clientId, layout } ) => {
```
Actually, this is the component that I find "useless" because it has no meaning on its own :) I would fine to just call `<BlockDraggable` in the block mover component.

nosolosw replied:
I find this is useful to hide the drag logic from the block-mover itself, which shouldn't be concerned with _how_ it is implemented. It helps the readers understand the general UI without overwhelming them with implementation details.

youknowriad reviewed (commented):
Looks good to me overall.

I personally like that the old way of showing which block is being dragged (the gray background at the position of the block being dragged). Maybe we could make it more clear which block is being dragged. It's not very obvious at the moment especially for text.

youknowriad commented:
@jasmussen A style "transform" like that should be doable yes :) the challenge would be which styles to apply and which ones to leave.

jasmussen commented:
For now, it would be interesting to try it just on the draggable clone.

I.e. if an editor style adds this:

```
body {
	background-color: blanchedalmond;
	color: firebrick;
}
```

... then it would be sweet if that could be transformed into:

```
body.gutenberg-editor-page,
.components-draggable__clone .editor-block-list__block-edit::before {
	background-color: blanchedalmond;
}
.components-draggable__clone .editor-block-list__block-edit {
	color: firebrick;
}
```

Sidenote, setting the text color on the body element doesn't seem to work in master, but that's still the plan right?

youknowriad commented:
I personally don't think we should do very specific transformations like these because we'll end up with a lot of specific cases and complexity.

 - You moved `background-color` to the `:before` of the draggable, what about ? `backgroud-*` properties or just `backgroud`? what about properties that touch the background but don't necessarily contain `background` in their name?
 - You moved `color` to the draggable but why? why not `font-size`, `text-transform`... and any other text properties?

I understand the benefits of this particular transform but want to ask if it's really worth it for the cost? For any one, of these transforms it's javascript code added, I'd like us to think more generically about those and avoid very specific rules like these.

> Sidenote, setting the text color on the body element doesn't seem to work in master, but that's still the plan right?

It should probably work yes, we need to figure out which rule is breaking the specificity in our CSS and move it to editor styles.

jasmussen commented:
> I understand the benefits of this particular transform but want to ask if it's really worth it for the cost? For any one, of these transforms it's javascript code added, I'd like us to think more generically about those and avoid very specific rules like these.

I understand this fully, but it's also the reason why we have no background on the draggable right now, which causes https://user-images.githubusercontent.com/1204802/45210024-cbda9480-b28e-11e8-9f34-6a726532c05d.png.

And although https://user-images.githubusercontent.com/1204802/45210605-57085a00-b290-11e8-8875-174162fe05da.png is not the end of the world (explicit white background), you can imagine if a theme set a white font color and we have a white background, then the text isn't visible.

The other alternative is to dim the draggable clone itself, make it a shadow, just like the slot it came from. But that kind of loses a lot of the magic doesn't it?

oandregal committed:
Get index and rootClientId from state

oandregal committed:
Rename arrows to icons

nosolosw commented on packages/editor/src/components/block-list/block.js:
```diff
@@ -509,12 +496,17 @@ export class BlockListBlock extends Component {
 						clientIds={ clientId }
 						rootClientId={ rootClientId }
 						layout={ layout }
+						order={ order }
```
021031c605953fdd69b013f16a24a1ceec09bfa5

nosolosw commented:
Rebased this with latest `master`.

mtias commented:
@jasmussen fwiw, I don't mind this:

<img width="330" alt="screen shot 2018-09-07 at 11 20 41" src="https://user-images.githubusercontent.com/1204802/45210605-57085a00-b290-11e8-8875-174162fe05da.png">

mtias commented:
In any case, I'd like for us to explore letting blocks rearrange as you drag as well separately.

jasmussen commented:
> fwiw, I don't mind this:

The problem is if the user is providing a dark editor style (see #9683) and white text. Then we'll have white text on a white background. I think it's worth looking at separately, I think @youknowriad might have a way to make this work, if we bribe him with the right flavor of milkshake.

nosolosw commented:
Rebased from `master` because a fix for an issue related to chrome and iframes has just landed https://github.com/WordPress/gutenberg/pull/9511 and wanted to make sure everything played well together.

@jasmussen @youknowriad would this PR be ready to merge? I'd love having it on 3.9. I understand the concerns related to styling the image clone, but I think that would merit its own separate PR - this doesn't change existing behavior.

Tested on Firefox and Chrome on Linux. Would love if I could get some testing on Mac as well. I know DnD is broken for Edge https://github.com/WordPress/gutenberg/issues/9729 and I don't expect this PR to fix it. I'll address that separately after I fix this infinite loop that affects children blocks https://github.com/WordPress/gutenberg/issues/9303

jasmussen commented:
Yes, I absolutely think this is ready to merge. GIF:

![dragondrop](https://user-images.githubusercontent.com/1204802/45422595-f6fb2480-b690-11e8-8cde-90636d30297b.gif)

It's really nice. The image clone thing is not a huge issue because the drag handle is now always visible. 

I think it would be nice to fix separately — adding the background color — if Riad can be convinced that's okay to do (see https://github.com/WordPress/gutenberg/pull/9569#issuecomment-419414645). But absolutely fine to do separately.

jasmussen commented:
It would be nice to separately also improve this:

![multiple drop zones](https://user-images.githubusercontent.com/1204802/45422690-40e40a80-b691-11e8-8cc8-7bda4e3859cd.gif)

There are 4 dropzones there, but there seems like there should only be two.

There's also a very obtuse issue, let me show more clearly.

![order](https://user-images.githubusercontent.com/1204802/45422759-7db00180-b691-11e8-8d27-d44e5eade3ad.gif)

In that there are two blocks, but notice how even though I'm dropping on a blue indicator that's _above_ the 2nd block, it lands _below_ the 2nd block.

mtias commented:
> It's really nice. The image clone thing is not a huge issue because the drag handle is now always visible.

And once blocks can rearrange themselves to make room for the dragged block, even better ;)

nosolosw commented:
Yeah, the dropping experience needs some love. Tracking this as separate issues:

- Consolidate consecutive drop zones https://github.com/WordPress/gutenberg/issues/9821
- Drop above a block moves it below instead https://github.com/WordPress/gutenberg/issues/9823
- Do not allow dropping below the blank inserter https://github.com/WordPress/gutenberg/issues/9824

youknowriad commented on packages/editor/src/components/block-mover/index.js:
```diff
@@ -106,15 +117,17 @@ export class BlockMover extends Component {
 			firstIndex: getBlockIndex( firstClientId, rootClientId ),
 			blockType: block ? getBlockType( block.name ) : null,
 			isLocked: getTemplateLock( rootClientId ) === 'all',
+			rootClientId,
```
This prop is useless right?

nosolosw replied:
Actually, no. It is used by `withDispatch`. It can't be retrieved there as `select` is not available within `withDispatch`, so it needs to be passed (previously it was a component prop passed by the parent).

youknowriad commented:
> There are 4 dropzones there, but there seems like there should only be two.

I think before this PR, The dropzones were styled in a way they show up exactly centered between two adjacent blocks, so the two dropzones overlap (the one after a block and the one before the next block), I consider this a small regression personally.

youknowriad commented:
Actually, I just checked in master and the adjacent dropzone have regressed elsewhere and not in this particular PR. It's fine to fix later.

Now, aside some useless props, this PR seems ready for me

youknowriad reviewed (approved):
LGTM 👍

aduth commented on packages/editor/src/components/block-mover/icons.js:
```diff
@@ -14,3 +14,11 @@ export const downArrow = (
 );
+
+export const dragHandle = (
+	<svg width="18" height="18" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 18 18" aria-hidden role="img" focusable="false">
```
Why is this not implemented as `AccessibleSVG` like the others in this file? As best I can tell, it doesn't conflict as being one. And if it did, we should leave an explaining comment why it ought to be different.

nosolosw replied:
As Riad said. Didn't know that existed. Going to update, test, and commit.

oandregal committed:
Use AccesibleSVG component

designsimply commented:
I noticed neither the draggable handle nor the movers are available in spotlight mode, and I wanted to check whether that is the intended behavior?

Every so tiny design nitpick: I noticed the icon for the drag handle (6 dots) is not the same shade of gray compared to the mover icons and it seemed to me they should be the same color. However, I also thought the color difference might intentional to make the draggable handle stand out a bit. ([screenshot](https://user-images.githubusercontent.com/1119271/45567806-035db980-b818-11e8-9139-e750ec724ee9.png))

youknowriad commented on packages/editor/src/components/block-list/style.scss:
```diff
@@ -1,104 +1,35 @@
-
-		.is-dark-theme & {
-			background-color: $light-opacity-light-200;
+.editor-block-list__layout .editor-block-list__block.is-selected { // Needs specificity to override inherited styles.
```
It looks like the changes here introduced a regression in Spotlight mode.

It should be fixed here https://github.com/WordPress/gutenberg/pull/9951
